sw/host/opentitanlib/src/image/manifest_def.rs

```diff
@@ -7,17 +7,27 @@ use crate::util::bigint::fixed_size_bigint;
 use crate::util::num_de::HexEncoded;
 use crate::util::parse_int::ParseInt;
 
-use anyhow::{anyhow, Result};
+use anyhow::{bail, Result};
 use serde::Deserialize;
 use std::convert::{TryFrom, TryInto};
 use std::iter::IntoIterator;
+use std::path::Path;
+use thiserror::Error;
 
-fixed_size_bigint!(ManifestRsa, 3072);
+use zerocopy::AsBytes;
 
-#[derive(Debug, Deserialize)]
+#[derive(Debug, Error)]
+pub enum ManifestError {
+    #[error("Manifest is missing field \"{0}\".")]
+    MissingField(&'static str),
+}
+
+fixed_size_bigint!(ManifestRsa, at_most 3072);
```

Comment:
nit: I guess `at_most` is for cases when these fields are not defined? Should we make them optional or use 0xff..ff instead of 0 rather than relaxing this constraint?

Reply:
The issue here is that a `ManifestDef` can come from the binary itself or from a deserailizable config file. In order for a `ManifestDef` to cleanly `try_into()` a `Manifest` all fields must be defined. We should be able to cleanly consume a `Manifest` from a binary and write it back without changing it, and since these fields are initialized to 0 in the image, we should be able to construct a `ManifestRsa` from a buffer of all 0's. An all 0 `ManifestRsa` is valid for converting back to a `Manifest` but a buffer of any other contents will fail if it is not exactly 96 32-bit words.

That does relax the constraint to allow bigints anywhere from 3041-3072 bits now that I think about it, since the MSB isn't checked. I could check this during the conversion to `Manifest` or use `[0xffffffff, ..., 0xffffffff]` as an intermediate to remove the need for the `at_most` qualifier.

It's a bit tricky since a `Manifest` cannot have undefined fields, but a `ManifestDef` can.